### ISSUE TITLE
Team rating features zero-fill teams from unrated competitions, so the model reads them as average

### ISSUE THREAD

**Issue by peteowen1** [peteowen1/panna#242]
## What

`02_player_ratings_to_team.R` zero-fills missing player ratings before aggregating to team level:

```r
ratings$psr[is.na(ratings$psr)] <- 0
if ("osr" %in% names(ratings)) ratings$osr[is.na(ratings$osr)] <- 0
if ("dsr" %in% names(ratings)) ratings$dsr[is.na(ratings$dsr)] <- 0
```

PSR is league-centred around 0, so a zero-filled player reads as **exactly average**. For a team whose entire squad is unrated, `home_sum_psr` is 0 — which the match model cannot distinguish from a genuinely median team.

## Why it matters

The teams this hits are not average. They are clubs from competitions panna does not rate — overwhelmingly Conference League opposition from small domestic leagues. They are systematically **weaker** than average, and the model is told they are typical.

This is the "a constant fill is INVISIBLE under fixed effects" failure mode: coverage % looks fine, the column is populated and correctly typed, and only the within-group variance would reveal it.

## Measurement (2026-09-12, from the cached `02_team_ratings.rds` / `04_match_dataset.rds`)

Current seasons: **101 of 8,595 team-matches (1.2%)** in 2025-2026 have `home_sum_psr == 0`. By competition:

| competition | zero-sum team-matches |
|---|---|
| UECL | 93 |
| UEL | 6 |
| AFCON | 2 |

Traced: of those 101, a median of **14 players appear in `match_stats`** and a median of **0** have a `seasonal_psr` row for that season. So it is a genuine coverage gap, **not** a join bug — the players were never rated, because their domestic league is not in the processed set.

Historically it was much worse, because the rated-league set was smaller. `pct_zero` = share of team-matches whose PSR sum is exactly 0; `n_rated` = median players the sum was built from (a full XI is 11):

| season | matches | n_rated | pct_zero | sd of feature |
|---|---|---|---|---|
| 2013 | 32 | 0 | 100.0 | 0.0000 |
| 2014 | 3,898 | 0 | 63.2 | 0.1161 |
| 2015 | 3,964 | 7 | 28.4 | 0.2617 |
| 2016 | 3,998 | 9 | 6.5 | 0.4391 |
| 2017 | 3,920 | 10 | 4.3 | 0.4978 |
| 2018 | 3,938 | 10 | 2.9 | 0.4830 |
| 2019 | 3,986 | 10 | 2.1 | 0.4970 |
| 2020-2026 | ~28,800 | 10 | 0.4-1.7 | 0.47-0.56 |

**Not PSR-specific.** The same zero-fill applies to the other rating features, and `panna` is affected more, not less — spread ratio 2020+ vs pre-2020 is 1.25x for `home_sum_psr`, 1.16x for `home_sum_epr`, **1.65x** for `home_sum_panna`.

## Effect size

Small and honest: ~1.2% of current team-matches, ~8% across all history, concentrated in 2014-2015 training data the model leans on least. This is a correctness matter, not a headline accuracy lever. Filing it so it is on the record rather than rediscovered.

## Suggested fix

Zero is a *fabricated value presented as a measurement*. Options, cheapest first:

1. **Add a coverage indicator feature** (e.g. `home_n_rated_players` already exists in `02_team_ratings.rds` but is not obviously carried into the model's feature set) so the model can learn "unrated squad" as its own signal rather than reading it as average.
2. **Pass NA instead of 0** and let the model's own missing handling deal with it (XGBoost handles NA natively), rather than asserting a value.
3. At minimum, **log the count per season** at the zero-fill site. Code that fabricates a value for a missing column should say so.

## How this surfaced

While building the held-out gate for PR #241, my team aggregation reproduced production at r=0.998 from 2020 on but only 0.65-0.73 before. Chasing that gap found these rows. My first framing of it ("the PSR feature is dead for 2015-2019") was **wrong** and is corrected above -- I had generalised from the handful of extreme rows I traced instead of measuring the distribution.

**Comment by peteowen1** [comment]
## Correction: the suggested Option 1 fix was already shipped

I never checked the trained model's actual feature list before writing "not obviously carried into the model's feature set" above. I should have.

Verified against the cached artifacts (\`04_match_dataset.rds\`, \`05_goals_model.rds\`): \`home_n_rated_players\`/\`away_n_rated_players\` are already present via \`04_build_match_dataset.R\`'s blanket \`left_join(team_ratings, ...)\`, and \`05_fit_goals_model.R\`'s feature selector auto-includes every numeric column not explicitly excluded -- so they were already model features, with zero code change needed.

They're genuinely contributing, not dead weight:

| model | feature | rank (of ~200) | gain |
|---|---|---|---|
| home-goals | \`away_n_rated_players\` | 38 | 0.0035 |
| home-goals | \`home_n_rated_players\` | 69 | 0.0020 |
| away-goals | \`home_n_rated_players\` | 41 | 0.0031 |
| away-goals | \`away_n_rated_players\` | 63 | 0.0022 |

Small next to \`elo_diff\` (0.308) but real, nonzero gain in both models.

## What's still true

The zero-fill mechanism itself is unchanged and still real -- a mostly-unrated squad's \`home_sum_psr\` is still a noisy, mis-centred number (traced example: a UECL side with 2 of 14 appearing players rated, \`home_sum_psr = -0.6315\`, more common than the "exactly 0" framing above suggested). But the model already has a compensating signal for it via \`n_rated_players\`, which the importance numbers confirm it uses.

## Also worth correcting

Option 2 ("pass NA instead of 0") in the original writeup is a **no-op** for \`home_sum_psr\` specifically -- \`sum(w * psr)\` has no \`na.rm\`, and under plain summation a zero-filled and an NA-excluded player contribute identically (both add nothing). It would matter for the \`avg_*\` columns (\`wmean\`), which aren't sum-invariant the same way, but not for the symptom this issue is about.

## Revised verdict

Given the fix already exists and is already earning real (if modest) gain, and the residual gap is ~1.2% of matches with a working partial mitigation already in place, this doesn't currently justify further model work. Leaving open as a record of the mechanism and the correction, not as an active TODO.

**Comment by peteowen1** [comment]
> *This was generated by AI during triage.*

## Triage Notes

Author's own final verdict (2026-09-13): the suggested fix (a coverage-indicator feature) was already shipped incidentally — `home_n_rated_players`/`away_n_rated_players` were already model features via a blanket join, confirmed contributing real (if modest) gain in both goals models. The residual zero-fill mechanism is real but small (~1.2% of current matches) and already partially compensated for.

Applying `wontfix` to reflect "no further action currently justified," per the author's explicit verdict — but NOT closing, since the author explicitly asked to leave this open as a record of the mechanism and the correction, not as an active TODO.
